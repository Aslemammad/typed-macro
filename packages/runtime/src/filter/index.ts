export { FilterOptions, FilterPattern, createFilter, Filter } from './filter'
