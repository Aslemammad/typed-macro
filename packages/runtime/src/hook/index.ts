export { createHookContainer, HookContainer, CombinedHooks } from './hook'
