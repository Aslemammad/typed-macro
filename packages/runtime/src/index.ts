export * from './transformer'
export * from './normalizer'
export * from './renderer'
export * from './runtime'
export * from './loader'
export * from './filter'
export * from './hook'
