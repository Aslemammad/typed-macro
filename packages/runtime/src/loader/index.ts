export { Loader, createLoader } from './loader'
