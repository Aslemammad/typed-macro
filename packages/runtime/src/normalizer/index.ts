export {
  ModularizedMacros,
  ModularizedModules,
  ModularizedTypes,
  NormalizedExports,
  normalizeExports,
  normalizeMacroProvider,
  NormalizedMacroProvider,
} from './normalizer'
