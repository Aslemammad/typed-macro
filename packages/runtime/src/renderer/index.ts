export { TypeRenderer, createTypeRenderer } from './renderer'
