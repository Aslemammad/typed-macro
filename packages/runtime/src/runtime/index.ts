export { Runtime, RuntimeOptions, createRuntime } from './runtime'
