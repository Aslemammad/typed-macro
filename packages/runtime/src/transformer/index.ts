export {
  Transformer,
  createTransformer,
  TransformerOptions,
} from './transformer'
