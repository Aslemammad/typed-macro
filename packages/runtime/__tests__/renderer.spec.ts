import { createTypeRenderer, TypeRenderer } from '../src'
import { readFile, writeFile } from 'fs/promises'
import { withTempPath } from './utils'

describe('TypeRenderer', () => {
  let renderer: TypeRenderer
  beforeEach(() => {
    renderer = createTypeRenderer()
  })

  it('should support append and render to string', () => {
    renderer.append('some', ['export type A = string'])
    expect(renderer.renderToString()).toMatchSnapshot()
    renderer.append('some', ['export type B = number'])
    renderer.append('other', ['export type A = string'])
    expect(renderer.renderToString()).toMatchSnapshot()
  })

  it('should support render to file', () => {
    renderer.append('some', ['export type A = string'])
    expect(async () => {
      await withTempPath(
        'not/exist/a.d.ts',
        async (tempPath) => await renderer.renderToFile(tempPath)
      )
    }).not.toThrow()
  })

  it('should truncate file before render', async () => {
    renderer.append('some', ['export type A = string'])
    await withTempPath('./a.d.ts', async (tempPath) => {
      await writeFile(tempPath, 'hello\n'.repeat(10))
      await renderer.renderToFile(tempPath)

      expect((await readFile(tempPath)).toString()).toBe(
        renderer.renderToString()
      )
    })
  })
})
