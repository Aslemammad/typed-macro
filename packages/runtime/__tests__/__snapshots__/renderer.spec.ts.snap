// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeRenderer should support append and render to string 1`] = `
"declare module 'some' {
export type A = string
}"
`;

exports[`TypeRenderer should support append and render to string 2`] = `
"declare module 'some' {
export type A = string
export type B = number
}
declare module 'other' {
export type A = string
}"
`;
